packages/react-composites/src/composites/MeetingComposite/ChatButtonWithUnreadMessagesBadge.tsx

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { ChatMessage } from '@azure/communication-chat';
+import { IStackStyles, Stack } from '@fluentui/react';
+import { Chat20Regular } from '@fluentui/react-icons';
+import { ControlBarButtonProps } from '@internal/react-components';
+import React, { useState } from 'react';
+import { useEffect } from 'react';
+import { ChatAdapter } from '../ChatComposite';
+import { ChatButton } from './ChatButton';
+import { NotificationIcon } from './NotificationIcon';
+
+/**
+ * @private
+ */
+export interface ChatButtonLogicWrapperProps extends ControlBarButtonProps {
+  chatAdapter: ChatAdapter;
+  isChatPaneVisible: boolean;
+}
+
+/**
+ * @private
+ */
+export const ChatButtonWithUnreadMessagesBadge = (props: ChatButtonLogicWrapperProps): JSX.Element => {
+  const { chatAdapter, isChatPaneVisible } = props;
+  const [unreadChatMessagesCount, setUnreadChatMessagesCount] = useState<number>(0);
+
+  /**
+   * Helper function to determine if the message in the event is a valid one from a user.
+   * Display name is used since system messages will not have one.
+   */
+  const validNewChatMessage = (message): boolean =>
```

Comment:
This doesn't capture any variables. So make it a module-level private function (just move it out as a `function`). Makes the code flow here easier to follow, and avoids crearting a new lambda on each render.

Reply:
resolved in this PR https://github.com/Azure/communication-ui-library/pull/1386 (oops I merged the wrong thing...)